File: app/Models/Wisata.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Wisata extends Model
{
    use HasFactory;

    protected $table = 'wisata';

    protected $fillable = [
        'user_id',
        'category_id',
        'name',
        'description',
        'location',
        'latitude',
        'longitude',
        'price',
        'opening_hours',
        'rating',
        'review_count',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function photos()
    {
        return $this->hasMany(Photo::class);
    }

    public function coverPhoto()
    {
        return $this->hasOne(Photo::class)->where('is_cover', true);
    }
    public function bookmarks()
    {
        return $this->hasMany(Bookmark::class);
    }
}

File: app/src/main/java/com/travelwaka/app/network/model/Wisata.kt
package com.travelwaka.app.network.model

data class Wisata(
    val id: Int,
    val user_id: Int,
    val category_id: Int,
    val name: String,
    val description: String,
    val location: String,
    val latitude: Double?,
    val longitude: Double?,
    val price: String,
    val opening_hours: String?,
    val rating: Float,
    val review_count: Int,
    val category: Category?,
    val photos: List<Photo>?,
    val cover_photo: Photo?,
    val created_at: String?
)

data class Category(
    val id: Int,
    val name: String,
    val icon: String?
)

data class Photo(
    val id: Int,
    val wisata_id: Int,
    val photo_url: String,
    val is_cover: Int  // ← ganti dari Boolean ke Int
)